### PR TITLE
Add Prometheus server

### LINKED ISSUE #1095: Add Prometheus Metrics Server
**Task summary**

Add Prometheus metrics server to the forest process.

**Specification reference**

The server should serialize the execution metrics and expose them on an HTTP server with a configurable port.

**Other information and links**

### PR DESCRIPTION
**Pull request by olibearo**
**Summary of changes**

Adds a Prometheus HTTP server to the `forest` process to support the collection of metrics. 
This PR does _not_ add metrics to `forest`, it only sets up the metrics scaffolding. 

**Reference issue to close (if applicable)**

Closes #1095 

**Other information and links**

### CODE REVIEW

File: forest/src/cli/mod.rs [hunk 1]
```diff
@@ -68,6 +68,12 @@ pub struct DaemonOpts {
     pub rpc: Option<bool>,
     #[structopt(short, long, help = "The port used for communication")]
     pub port: Option<String>,
+    #[structopt(
+        short,
+        long,
+        help = "Port used for Prometheus metric collection server"
+    )]
+    pub prometheus_port: Option<u16>,
```
Comment (ec2) [inline forest/src/cli/mod.rs:76]: I'd call this `metrics_port`. Would be easier to people to understand what it is if they dont know prometheus. Or if we move away from prometheus later, the flag doesnt need to change

Comment (ec2) [reply forest/src/cli/mod.rs:76]: (Thats a nit btw, so dont worry about it)

**Review by ec2 — APPROVED** [review]
Remember to run the license script :p

**Review by creativcoder — APPROVED** [review]

File: forest/src/daemon.rs [hunk 2]
```diff
@@ -2,27 +2,29 @@
...
 
+use async_std::{channel::bounded, sync::RwLock, task};
+use libp2p::identity::{ed25519, Keypair};
+use log::{debug, info, trace};
+use rpassword::read_password;
+
+use std::io::prelude::*;
+use std::path::PathBuf;
+use std::sync::Arc;
+
```
Comment (cryptoquick) [inline forest/src/daemon.rs:27]: Thank you for separating these out! I usually do them in the opposite order, though, but either way, it really helps vs mixing them in. I also do similar in Cargo.toml and use `#` to separate them.

Comment (olibearo) [reply forest/src/daemon.rs:27]: Yeah, I would really like it if this was part of our style guide

**Pull request merged**